### ISSUE TITLE
Understanding how to move the offset for a displacementFilter

### ISSUE THREAD

**Issue by Dairnon** [pixijs/pixi.js#2923]
I'm currently using RPG Maker MV (which uses pixi) to apply a displacementFilter to a container of sprites. I've gotten the displacementFIlter working but am having issues with getting it to allow me to move/offset the texture on the sprite. 

I was using the "TilingSprite" object to create a tillable sprite which I can offset via .origin.x (which works for regular tilable sprites) but that doesn't seem to be working for the filter.  Since I know a few developers behind pixi have worked with PIXI for RPGMaker MV I was hoping someone had any tips or idea how to move this displacementFilter. 

I've been able to get the filter to move by moving the entire sprite, but that causes the sprite to move out of frame.

**Comment by ivanpopelyshev** [comment]
Hey! For filter, you hae to specify the texture, right? And for texture, in pixiv4, you can do the following:

```js
texture.baseTexture.wrapMode = PIXI.WRAP_MODES.REPEAT;
```

in RPGMV baseTexture is somewhere inside the bitmap:

```js
bitmap._baseTexture.wrapMode = PIXI.WRAP_MODES.REPEAT;
```

or something like that, please "console.log" the bitmap and look inside, where is it.

You have to do that BEFORE it renders the first time!

**Comment by Dairnon** [comment]
Thanks so much for the quick reply. Before it renders the first time? Where exactly does it do that in RPG Maker MV? So I know where to act. 

(Again THANKS SO MUCH FOR YOUR HELP)

**Comment by ivanpopelyshev** [comment]
When you create that filter and assign texture to it, i think it will be fine :)

**Comment by Dairnon** [comment]
I think its working! Thanks so much Ivan. Your the best (along with all the other PIXI devs) 

Btw side question before closing... do you know if that memory leak is fixable and or will be fixed for MV? It's very worrying.

**Comment by ivanpopelyshev** [comment]
I'm working on it. Yeah, there's that tilingSprite leak.

**Comment by Dairnon** [comment]
Is the leak where they didn't destroy the textures still in? Or were you able to fix that for 1.3. I remember reading a forum post of you mentioning that. 

I was also curious as to if these problems have to do with locking up the game when transferring between rooms. (after about 10  - 20 transfers, the game will slightly freeze / lock up for a second or two) That has been very worrying as it ruins pacing.  I'd rather a fix for it make overall transitioning slower, but prevent the locking up/ freezing. 

Again thanks for the quick responses.

**Comment by ivanpopelyshev** [comment]
It is destroying the textures. Problem is that RPGMV is stashing EVERYTHING, and sometimes in x3 times. If you have 500MB of textures, it will stash about 1.5GB in the end, and it wont grow after that.

**Comment by Dairnon** [comment]
Alright that's interesting to know. Hopefully this lagging problem when changing so many maps gets fixed. And maybe the devs stop storing textures 3x to lower the memory footprint lol. 

Thanks so much for your time and support!

**Comment by Dairnon** [comment]
Okay one last thing (very sorry.) 

At the start of a map, the displacement filter creates whatever its supposed to filter to become black for a second. Is there a way to prevent this? Thanks!

**Comment by ivanpopelyshev** [comment]
As for blur, you can try latest release of pixi, I remember that something was changed in it regarding the filters. RPGMV should work with it.

**Comment by Dairnon** [comment]
Okay! So you think this will remove the weird flickering/black flashing of the displacement filter and launch too? And I was curious as to how to install this. The last time I tried adding a new version of PIXI I had a lot of problems getting that to work, even lately when  I tried adding new PIXI v4 filters to the project I had no idea how to go about getting those to work either. 

Is it as simple as replacing the old PIXI file with the new one?

**Comment by Dairnon** [comment]
I also may have found a bug or I used the filter wrong. I applied the filter to a somewhat transparent background parallax as well as to a container holding copies of sprites in the game, upside down and used for reflections. 

For some strange reason if I walk between two distances the filter jumps to a weird position. (In this example my sprite used for the displacement isn't moving at all)

https://youtu.be/ZZisu6yJUpw
